src/util/rbs_allocator.c

```diff
@@ -139,52 +122,23 @@ void *rbs_allocator_realloc_impl(rbs_allocator_t *allocator, void *ptr, size_t o
 // Allocates `size` bytes from `allocator`, aligned to an `alignment`-byte boundary.
 void *rbs_allocator_malloc_impl(rbs_allocator_t *allocator, size_t size, size_t alignment) {
     rbs_assert(size % alignment == 0, "size must be a multiple of the alignment. size: %zu, alignment: %zu", size, alignment);
-
-    if (default_page_payload_size < size) { // Big allocation, give it its own page.
-        // How much we need to pad the new page's payload in order to get an aligned pointer
-        // hack?
-        const size_t alignment_padding = needed_padding((void *) (sizeof(rbs_allocator_page_t) + size), alignment);
-
-        rbs_allocator_page_t *new_page = rbs_allocator_page_new_large(alignment_padding + size);
-
-        // This simple allocator can only put small allocations into the head page.
-        // Naively prepending this large allocation page to the head of the allocator before the previous head page
-        // would waste the remaining space in the head page.
-        // So instead, we'll splice in the large page *after* the head page.
-        //
-        // +-------+    +-----------+        +-----------+
-        // | arena |    | head page |        | new_page  |
-        // |-------|    |-----------+        |-----------+
-        // | *page |--->|  size     |   +--->|  size     |   +---> ... previous tail
-        // +-------+    |  offset   |   |    |  offset   |   |
-        //              | *next ----+---+    | *next ----+---+
-        //              |    ...    |        |    ...    |
-        //              +-----------+        +-----------+
-        //
-        new_page->next = allocator->page->next;
-        allocator->page->next = new_page;
-
-        return new_page->payload + alignment_padding;
-    }
-
-    void *p = rbs_allocator_page_attempt_alloc(allocator->page, size, alignment);
-    if (p != NULL) return p;
-
-    // Not enough space. Allocate a new page and prepend it to the allocator's linked list.
-    rbs_allocator_page_t *new_page = rbs_allocator_page_new_default();
-    new_page->next = allocator->page;
-    allocator->page = new_page;
-
-    p = rbs_allocator_page_attempt_alloc(new_page, size, alignment);
-    rbs_assert(p != NULL, "Failed to allocate a new allocator page");
-    return p;
+    uintptr_t aligned = align(allocator->heap_ptr, alignment);
+    allocator->heap_ptr = aligned + size;
+    return (void *) aligned;
 }
 
 // Note: This will eagerly fill with zeroes, unlike `calloc()` which can map a page in a page to be zeroed lazily.
 //       It's assumed that callers to this function will immediately write to the allocated memory, anyway.
 void *rbs_allocator_calloc_impl(rbs_allocator_t *allocator, size_t count, size_t size, size_t alignment) {
     void *p = rbs_allocator_malloc_many_impl(allocator, count, size, alignment);
+#ifdef _WIN32
     memset(p, 0, count * size);
+#elif !defined(NDEBUG)
```

Comment:
I don't think we _ever_ set `NDEBUG` (that's something to fix, I guess), so this is gonna be pretty pricy.